src/runtime/propensity-server-test.js

```diff
@@ -296,4 +312,132 @@ describes.realWin('PropensityServer', {}, env => {
           expect(() => {throw reason;}).to.throw(/Invalid request/);
         });
   });
+
+  it('should listen for events from event manager', function*() {
+    //event registration happens during initialization so we need to stub it
+    //and then create a new propensity server
+    const eventMan = new ClientEventManager();
+    let receivedEvent = null;
+    sandbox.stub(PropensityServer.prototype, 'sendClientEvent_', event => {
+      receivedEvent = event;
+    });
+    new PropensityServer(win, pubId, eventMan);
+    eventMan.logEvent(defaultEvent);
+    yield eventMan.lastAction_;
+    expect(defaultEvent === receivedEvent).to.be.true;
+  });
+
+  it('should convert analytics events to propensity events', () => {
+    let receivedType = null;
+    let receivedContext = null;
+    sandbox.stub(PropensityServer.prototype, 'sendEvent',
```

Comment:
I see you are trying to test the mapping here. Since sendEvent_ is going to be a private method and nobody outside is calling it, testing arguments passed in between private methods inside a class amounts to testing implementation. We need to test behavior. 

Do you need to test the mapping?

Reply:
I initially thought I didn't but then later thought it would be better if I did.  I'm not sure - do I need to test the mapping?    I lean towards yes in case someone accidentally changes it but I'm not really sure.

Reply:
as discussed, lets create another class to centralize event mapping across different entities that use it.

Reply:
Done!